src/js/actions/libraries.js

```diff
@@ -242,53 +279,62 @@ define(function (require, exports) {
             currentLayers = currentDocument.layers.selected,
             currentLayer = currentLayers.first();
 
-        if (!currentLibrary || currentLayers.size !== 1 ||
+        if (!currentLibrary ||
+            currentLayers.size !== 1 ||
             !currentLayer || !currentLayer.isTextLayer()) {
             return Promise.resolve();
         }
 
-        // FIXME: All constants like this should be described in one location for later
-        var CHARACTERSTYLE_TYPE = "application/vnd.adobe.element.characterstyle+dcx",
-            REPRESENTATION_TYPE = "application/vnd.adobe.characterstyle+json";
-
-        currentLibrary.beginOperation();
-
-        // FIXME: We should build this name correctly, instead of layer name
-        var newElement = currentLibrary.createElement(currentLayer.name, CHARACTERSTYLE_TYPE),
-            representation = newElement.createRepresentation(REPRESENTATION_TYPE, "primary"),
-            imageRepresentation = newElement.createRepresentation("image/png", "rendition"),
-            typeData = fontStore.getTypeObjectFromLayer(currentLayer),
-            // FIXME: Mac/Win temporary locations!
-            filepath = "/tmp/textThumbnailPreview.png";
-
-        // Where magic happens
-        representation.setValue("characterstyle", "data", typeData);
-
-        // FIXME: Make sure this reflects the character style we're recreating
-        // check to see how CEP Panel does it.
-        var exportObj = libraryAdapter.createTextThumbnail(filepath,
-            typeData.adbeFont.postScriptName,
-            "Aa",
-            36,
-            colorAdapter.colorObject([0, 0, 0]));
-
-        return descriptor.playObject(exportObj)
-            .bind(this)
-            .then(function () {
-                return Promise.fromNode(function (cb) {
-                    imageRepresentation.updateContentFromPath(filepath, false, cb);
-                });
+        var typeData = fontStore.getTypeObjectFromLayer(currentLayer),
+            tempPreviewPath;
+        
+        return _getTempPaths().bind(this)
+            .then(function (paths) {
+                // Create Character Style preview 
+                
+                tempPreviewPath = paths.tempPreviewPath;
+                
+                // FIXME: Make sure this reflects the character style we're recreating
```

Comment:
Can you track this FIXME in an issue? If we are happy with how we're rendering the character style, lets' get rid of this FIXME comment.